README.md

```diff
@@ -1,89 +1,92 @@
-      "balances": [
-        "0x000000000000000000000000000000000000000000000000000000000001375f",
-        "0x000000000000000000000000000000000000000000000000000000000000e9a6"
-      ]
-    },
-    "nodes": [
-      [
-        {
-          "hash": "0x0e113acd03b98f0bab0ef6f577245d5d008cbcc19ef2dab3608aa4f37f72a407",
-          "balances": [
-            "0x0000000000000000000000000000000000000000000000000000000000002e70",
-            "0x000000000000000000000000000000000000000000000000000000000000a0cb"
-          ]
-        },
-        {
-          "hash": "0x17ef9d8ee0e2c8470814651413b71009a607a020214f749687384a7b7a7eb67a",
-          "balances": [
-            "0x00000000000000000000000000000000000000000000000000000000000108ef",
-            "0x00000000000000000000000000000000000000000000000000000000000048db"
-          ]
-        }
-      ],
-      [
-        {
-          "hash": "0x2a4a7ae82b45b3800bdcd6364409e7ba9cac3d4598c546bd48952c234b5d2fb9",
-          "balances": [
-            "0x000000000000000000000000000000000000000000000000000000000001375f",
-            "0x000000000000000000000000000000000000000000000000000000000000e9a6"
-          ]
-        }
-      ]
-    ],
-    "depth": 1,
-    "entries": [
-      {
-        "balances": [
-          "11888",
-          "41163"
-        ],
-        "username": "dxGaEAii"
-      },
-      {
-        "balances": [
-          "67823",
-          "18651"
-        ],
-        "username": "MBlfbBGI"
-      }
-    ],
-    "is_sorted": false
-  }
-  ```
+Summa Aggregation is a scalable solution specifically designed to accelerate the process of building Merkle sum tree. It addresses the time-intensive challenge of constructing these trees by enabling efficient scaling through parallelization and distributed computation across multiple machines.
 
+## Running test
+
+Tests can be run using the following command:
+
+```bash
+cargo test --release
+```
+
+Note: The Worker will run locally and uses port 4000 as the default for its server.
+Please ensure that this port is not already in use to avoid errors.
+
+## Running Additional Tests Involving Docker and Docker Swarm
+
+For additional tests involving Docker and Docker Swarm mode, the presence of the "summadev/summa-aggregation-mini-tree" image in the local Docker registry is required.
+
+
+### Building the docker image
+
+Build the image using the following command:
+
+```bash
+docker build . -t summadev/summa-aggregation-mini-tree
+```
+
+### Downloading the Docker Image
+
+Alternatively, the image can be downloaded from Docker Hub:
+
+```bash
+docker pull summadev/summa-aggregation-mini-tree
+```
+
+### Testing with LocalSpawner
+
+The following command runs an additional test case using the LocalSpawner, which spawns worker containers in the local Docker environment. This extra test case involves running two containers during the testing process:
+
+
+```bash
+cargo test --features docker
```

Comment:
I encounter the following error when running this test. Is it because of my operative system? If so, I would specify it in the comments. Other tests `cargo test --release` are passing correctly

`---- orchestrator::test::test_with_containers stdout ----
docker-info: SystemInfo { id: Some("5849d88f-af31-4b4d-89ac-95cb55fcd186"), containers: Some(0), containers_running: Some(0), containers_paused: Some(0), containers_stopped: Some(0), images: Some(5), driver: Some("overlay2"), driver_status: Some([["Backing Filesystem", "extfs"], ["Supports d_type", "true"], ["Using metacopy", "false"], ["Native Overlay Diff", "true"], ["userxattr", "false"]]), docker_root_dir: Some("/var/lib/docker"), plugins: Some(PluginsInfo { volume: Some(["local"]), network: Some(["bridge", "host", "ipvlan", "macvlan", "null", "overlay"]), authorization: None, log: Some(["awslogs", "fluentd", "gcplogs", "gelf", "journald", "json-file", "local", "logentries", "splunk", "syslog"]) }), memory_limit: Some(true), swap_limit: Some(true), kernel_memory_tcp: None, cpu_cfs_period: Some(true), cpu_cfs_quota: Some(true), cpu_shares: Some(true), cpu_set: Some(true), pids_limit: Some(true), oom_kill_disable: Some(false), ipv4_forwarding: Some(true), bridge_nf_iptables: Some(true), bridge_nf_ip6tables: Some(true), debug: Some(false), nfd: Some(47), n_goroutines: Some(65), system_time: Some("2023-12-07T10:31:08.580451711Z"), logging_driver: Some("json-file"), cgroup_driver: Some(CGROUPFS), cgroup_version: Some(_2), n_events_listener: `